pkg/database/plugin.go

```diff
@@ -104,6 +105,12 @@ type iMessageCollection interface {
 
 	// GetMessagesForData - List messages where there is a data reference to the specified ID
 	GetMessagesForData(ctx context.Context, dataID *fftypes.UUID, filter Filter) (message []*fftypes.Message, res *FilterResult, err error)
+
+	// GetMessageBatchIDs - an optimized query to retrieve any non-null batch IDs for a list of message IDs
+	GetMessageBatchIDs(ctx context.Context, msgIDs []driver.Value) (batchIDs []*fftypes.UUID, err error)
```

Comment:
Can we not use `[]*fftypes.UUID` for the IDs?

Reply:
It means an extra loop and array allocation - I don't mind doing it here as it's a component boundary interface.

Reply:
(sorry comments here and above are a bit interleaved)